### PR TITLE
Add PreCreateContainerHook and PostStopSandboxHook and update parameters

### PR DESCRIPTION
**Pull request by cheimu**
Signed-off-by: yimo <yimo@xiaohongshu.com>

### Ⅰ. Describe what this PR does
This PR 
1. proposes add two more interfaces within protocol between runtime proxy and its hook server, including PreCreateContainerHook, which could set container resources setting before creating, and PostStopSandboxHook, which could do the resource setting garbage collecting before pods deleted.
2. Unified pod-level request/response name
3. Add `cgroup_parent` in container-level request/response body

### CODE REVIEW

**Comment by cheimu** [comment]
@ZYecho  @honpey @songzh215

File: apis/runtime/v1alpha1/api.pb.go [hunk 1]
```diff
@@ -14,22 +14,36 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+//
```
Comment (jasonliu747) [inline apis/runtime/v1alpha1/api.pb.go:17]: duplicate copyright header

File: apis/runtime/v1alpha1/api.proto [hunk 2]
```diff
@@ -127,6 +127,7 @@ message ContainerResourceHookRequest {
   // pod related annotations and labels
   map<string, string> pod_annotations = 6;
   map<string, string> pod_labels = 7;
+  string cgroup_parent = 8;
```
[the remark this answers is not in the document]
Comment (cheimu) [reply apis/runtime/v1alpha1/api.proto]: Done

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Pull request merged**